src/Compilers/CSharp/Test/Semantic/Semantics/TopLevelStatementsTests.cs

```diff
@@ -8907,10 +8856,420 @@ class Program
+            var text = @"
+M();
+
+public class Program
+{
+    private static void M()
+    {
+        System.Console.WriteLine(""Hi!"");
+    }
+}
+";
+
+            var comp = CreateCompilation(text, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
+            comp.VerifyDiagnostics(
+                // (4,14): error CS0260: Missing partial modifier on declaration of type 'Program'; another partial declaration of this type exists
+                // public class Program
+                Diagnostic(ErrorCode.ERR_MissingPartial, "Program").WithArguments("Program").WithLocation(4, 14)
                 );
+            var entryPoint = SynthesizedSimpleProgramEntryPointSymbol.GetSimpleProgramEntryPoint(comp);
+            Assert.Equal("System.Void", entryPoint.ReturnType.ToTestDisplayString());
+            Assert.True(entryPoint.ReturnsVoid);
+            AssertEntryPointParameter(entryPoint);
+            Assert.Same(entryPoint, comp.GetEntryPoint(default));
+            Assert.False(entryPoint.CanBeReferencedByName);
+            Assert.True(entryPoint.ContainingType.CanBeReferencedByName);
+            Assert.Equal("<Main>$", entryPoint.Name);
+            Assert.Equal("Program", entryPoint.ContainingType.Name);
+            Assert.Equal(Accessibility.Public, entryPoint.ContainingType.DeclaredAccessibility);
+            Assert.Equal(Accessibility.Public, entryPoint.DeclaredAccessibility);
+        }
+
+        [Fact]
+        public void SpeakableEntyrPoint_ProgramIsInternal()
+        {
+            var text = @"
+M();
+
+internal partial class Program
+{
+    private static void M()
+    {
+        System.Console.WriteLine(""Hi!"");
+    }
+}
+";
+
+            var comp = CreateCompilation(text, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
+            CompileAndVerify(comp, expectedOutput: "Hi!");
+            var entryPoint = SynthesizedSimpleProgramEntryPointSymbol.GetSimpleProgramEntryPoint(comp);
+            Assert.Equal(Accessibility.Internal, entryPoint.ContainingType.DeclaredAccessibility);
+            Assert.Equal(Accessibility.Public, entryPoint.DeclaredAccessibility);
+        }
+
+        [Fact]
+        public void SpeakableEntyrPoint_ProgramWithoutDeclaredAccessibility()
+        {
+            var text = @"
+M();
+
+partial class Program
+{
+    private static void M()
+    {
+        System.Console.WriteLine(""Hi!"");
+    }
+}
+";
+
+            var comp = CreateCompilation(text, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
+            CompileAndVerify(comp, expectedOutput: "Hi!");
+            var entryPoint = SynthesizedSimpleProgramEntryPointSymbol.GetSimpleProgramEntryPoint(comp);
+            Assert.Equal(Accessibility.Internal, entryPoint.ContainingType.DeclaredAccessibility);
+            Assert.Equal(Accessibility.Public, entryPoint.DeclaredAccessibility);
+        }
+
+        [Fact]
+        public void SpeakableEntyrPoint_ProgramIsStruct()
```

Comment:
>ProgramIsStruct [](http://example.com/codeflow?start=40&length=15)

Consider testing `partial interface Program`. And it looks like that test, along with the `struct` and `record` tests here could be combined with `[Theory]`.

Reply:
Interfaces would require a different test because of static method `M`... Not sure it's worth the trouble